Subject: compile error fix (needs review)
From: Sven-Thorsten Dietrich

RT changes __list_splice to require prev and next pointers
to be passed in. 

This changes the use in the new code to list_splice_tail, 
but the optimal insertion point needs to be analyzed.

Signed-off-by: Sven-Thorsten Dietrich <sdietrich@suse.de>

---
 drivers/dma/ioat_dma.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23/drivers/dma/ioat_dma.c
===================================================================
--- linux-2.6.23.orig/drivers/dma/ioat_dma.c
+++ linux-2.6.23/drivers/dma/ioat_dma.c
@@ -244,7 +244,7 @@ static dma_cookie_t ioat_tx_submit(struc
 	/* write address into NextDescriptor field of last desc in chain */
 	to_ioat_desc(ioat_chan->used_desc.prev)->hw->next =
 							first->async_tx.phys;
-	__list_splice(&new_chain, ioat_chan->used_desc.prev);
+	list_splice_tail(&new_chain, ioat_chan->used_desc.prev);
 
 	ioat_chan->pending += desc_count;
 	if (ioat_chan->pending >= 4) {
